gerritlab/main.py

```diff
@@ -205,26 +207,70 @@ def create_merge_requests(repo, remote, local_branch):
                 title=title,
                 description=desp,
             )
+            proposed_new_mrs.append(mr)
             c.mr = mr
-            with timing("create_mrs"):
-                mr.create()
-                mr._sha = None
-            new_mrs.append(mr)
 
     # At this point we have one MR for each commit.
     # title/desc/target_branch may be out-of-date for preexisting MRs.
     # Augment the MR descriptions to include a list of related MRs.
     for c in commits_data:
         title, desc = generate_augmented_mr_description(commits_data, c)
         mr = c.mr
-
         mr.set_target_branch(c.target_branch)
         mr.set_title(title)
         mr.set_desc(desc)
-        with timing("update_mrs"):
-            if (mr.save() or mr._sha != c.commit.hexsha) and mr not in new_mrs:
-                updated_mrs.append(mr)
-                commits_to_pipeline_cancel.append(c.commit)
+        if mr not in proposed_new_mrs:
```

Comment:
For this scenario:

Commit A followed by commit B.  
git lab
edit commit B
git lab

this code will report that the MR for commit A will be updated though nothing about it changed.